### PR TITLE
samples: openthread: Enable USB CDC ACM for Spinel

### PR DESCRIPTION
**Pull request by lmaciejonczyk**
It enables emulation serial port over USB in Co-processor sample.
The communication host - co-processor is done through USB.

### CODE REVIEW

**Comment by lmaciejonczyk** [comment]
@ru-fu Feel free to modify docs changes if needed, thanks.

File: samples/openthread/common/overlay-ot-defaults.conf [hunk 1]
```diff
@@ -4,8 +4,8 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 
-CONFIG_NCS_SAMPLES_DEFAULTS=y
```
Comment (edmont) [inline samples/openthread/common/overlay-ot-defaults.conf:7]: Why this change? I guess we should keep logging enabled as default for the other OT samples. Also the lack of these may be the cause of the code size increase for the CoAP client sample:
```
	imply ASSERT_NO_COND_INFO
	imply ASSERT_NO_MSG_INFO
```

Comment (lmaciejonczyk) [reply samples/openthread/common/overlay-ot-defaults.conf:7]: I removed this because it imply LOG_MINIMAL which causes build warning for NCP/RCP. We want to have deferred logging especially for NCP/RCP to decrease logging impact on the application. I cannot overwrite LOG_MINIMAL in the sample overlay since it's promptless option. What more I belive we should use LOG_MODE_DEFFERED for all samples apart from minimal configuration. It fixes NCP/RCP build where we tried to disable LOG_MINIMAL but it has been set anyway with the build warning.

**Review by rlubos — APPROVED** [review]

File: samples/openthread/common/overlay-ot-defaults.conf (hunk 1 above)
Comment (edmont) [reply samples/openthread/common/overlay-ot-defaults.conf:7]: Fair enough.

File: doc/nrf/ug_thread_tools.rst [hunk 2]
```diff
@@ -274,16 +274,17 @@ The interaction is possible using commands proper to wpanctl, a module installed
 Installing wpantund
 ===================
 
-To ensure that the interaction with the samples works as expected, install the version of wpantund that has been used for testing the |NCS|.
```
Comment (ru-fu) [inline doc/nrf/ug_thread_tools.rst:277]: Why are we getting rid of the general installation and version instructions? They were requested not long ago.
We can still have a note stating that you need a different version for USB.

File: doc/nrf/ug_thread_tools.rst [hunk 3]
```diff
@@ -310,23 +311,13 @@ When working with samples that support wpantund, complete the following steps to
 
         wpantund -I *network_interface_name* -s *ncp_uart_device* -b *baud_rate*
 
-     For example::
+     For example (UART transport)::
 
         sudo wpantund -I leader_if -s /dev/ttyACM0 -b 1000000
 
-   Radio co-processor (RCP)
```
Comment (ru-fu) [inline doc/nrf/ug_thread_tools.rst:317]: So we cannot do this for RCP anymore?

**Review by edmont — APPROVED** [review]

File: doc/nrf/ug_thread_tools.rst [hunk 4]
```diff
@@ -406,23 +395,6 @@ When working with samples that support Pyspinel, complete the following steps to
...
-
-        sudo python3 spinel-cli.py -d *debug_level* -p './output/posix/bin/ot-ncp spinel+hdlc+uart://\ *ncp_uart_device*\ ?uart-baudrate=\ *baud_rate*
-
-     For example::
-
-        sudo python3 spinel-cli.py -d 4 -p './output/posix/bin/ot-ncp spinel+hdlc+uart:///dev/ttyACM0?uart-baudrate=1000000'
-
 .. _ug_thread_tools_ot_apps:
 
 OpenThread POSIX applications
```
Comment (ru-fu) [inline doc/nrf/ug_thread_tools.rst:400]: So you must now use wpantund for NCP and ot-cli/ot-daemon for RCP?

**Review by ru-fu — CHANGES_REQUESTED** [review]

File: doc/nrf/ug_thread_tools.rst (hunk 3 above)
Comment (lmaciejonczyk) [reply doc/nrf/ug_thread_tools.rst:317]: That's correct. ot-ncp application, which we were using for connecting to RCP in wpantund, has been removed and is not evailable any more.

File: doc/nrf/ug_thread_tools.rst (hunk 2 above)
Comment (lmaciejonczyk) [reply doc/nrf/ug_thread_tools.rst:277]: Update the wpantund commit to: bf45115f41ba2b8029eda174be2b93dea73b9261 please

File: doc/nrf/ug_thread_tools.rst [hunk 5]
```diff
@@ -451,29 +426,68 @@ Building the OpenThread POSIX applications
...
-#. Build the applications with the required options.
-   For example, to build POSIX applications like ``ot-cli`` or ``ot-ncp`` with log output being redirected to the application, run the following command:
+   Alternatively, to build the ``ot-daemon`` and ``ot-ctl`` applications with support for USB transport and Thread v1.1, run the following command::
+
+      ./script/cmake-build posix -DOT_SPINEL_RESET_CONNECTION=ON -DOT_THREAD_VERSION=1.1 -DOT_DAEMON=ON
+
+You can find the generated applications in :file:`./build/posix/src/posix/`.
+
+.. note::
+   The build option ``OT_SPINEL_RESET_CONNECTION`` is required to properly handle USB communication with RCP devices that perform a hard reset.
```
Comment (ru-fu) [inline doc/nrf/ug_thread_tools.rst:445]: You had NCP/RCP here first, but I thought we're using this for RCP only?

**Review by ru-fu — COMMENTED** [review]
I added a commit - please check and squash, if okay.

File: doc/nrf/ug_thread_tools.rst (hunk 5 above)
Comment (lmaciejonczyk) [reply doc/nrf/ug_thread_tools.rst:445]: Yes, OT posix applications are for RCP only.

**Comment by lmaciejonczyk** [comment]
> I added a commit - please check and squash, if okay.

LGTM, thanks!

File: samples/openthread/coprocessor/README.rst [hunk 6]
```diff
@@ -128,7 +128,7 @@ The following configuration files are available:
 * :file:`overlay-minimal_rcp.conf` - Enables a minimal configuration that reduces the code size and RAM usage.
   This file enables the RCP architecture with basic functionality and optimizes stacks and buffer sizes.
   For more information, see :ref:`app_memory`.
-
+* :file:`overlay-usb.conf` - Enables emulation serial port over USB for Spinel communication with the host.
```
Comment (ru-fu) [inline samples/openthread/coprocessor/README.rst:131]:
```suggestion
* :file:`overlay-usb.conf` - Enables emulating a serial port over USB for Spinel communication with the host.
```

**Review by ru-fu — COMMENTED** [review]
Sorry, I missed one file.

**Review by ru-fu — APPROVED** [review]
